### PR TITLE
Added HNS Encryption Context

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
In progress

File: sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakePathCreateOptions.cs [hunk 1]
```diff
@@ -57,7 +57,9 @@ public class DataLakePathCreateOptions
 
         /// <summary>
         /// Optional encryption context that can be set the file.
-        /// Encryption context is intended to store metadata that can be used to decrypt the blob.
+        /// Encryption context is file metadata that is not encrypted when stored on the file.
+        /// The primiary application of this field is to store non-encrypted data that can be used to derive the customer-provided key
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakePathCreateOptions.cs:61]: NIT - spelling primiary => primary

Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakePathCreateOptions.cs:61]: ctrl-f and replace for this, since it's copy pasted elsewhere

**Review by amnguye — APPROVED** [review]
LGTM - just remember to change back the `autorest.md` file.

File: sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakePathCreateOptions.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/Models/DataLakePathCreateOptions.cs]: Fixed.

**Pull request merged**